internal/pkg/cli/deploy/workload.go

```diff
@@ -311,40 +345,52 @@ func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPus
 		return nil, nil
 	}
 	// If it is built from local Dockerfile, build and push to the ECR repo.
-	buildArg, err := buildArgs(d.name, d.imageTag, d.workspacePath, d.mft)
+	buildArgsPerContainer, err := buildArgsPerContainer(d.name, d.workspacePath, d.image, d.mft)
```

Comment:
nit: its just my preference which looks short and sweet. 
```suggestion
	buildArgsFor, err := buildArgsPerContainer(d.name, d.workspacePath, d.image, d.mft)
```